hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -260,6 +260,17 @@ func validateHostedClusterUpdate(new *hyperv1.HostedCluster, old *hyperv1.Hosted
 		old.Spec.Networking.APIServer.Port = new.Spec.Networking.APIServer.Port
 	}
 
+	// We default the basedomain for KubeVirt clusters, so we allow the baseDomain
+	// to go from unset to set, but no updates after basedomain is set.
+	baseDomainPassthrough := new.Annotations[hyperv1.KubeVirtBaseDomainPassthroughAnnotation]
+	if new.Spec.Platform.Type == hyperv1.KubevirtPlatform &&
+		baseDomainPassthrough == "true" &&
+		new.Spec.DNS.BaseDomain != "" &&
+		old.Spec.DNS.BaseDomain == "" {
+
+		old.Spec.DNS.BaseDomain = new.Spec.DNS.BaseDomain
+	}
+
```

Comment:
BaseDomain is typically immutable. I had to make an exception when the BaseDomain is empty and the kubevirt platform is in use. This is how we default the BaseDomain.